### PR TITLE
DM-39857: Stop using --flake8 for tests and add ruff configuration

### PR DESCRIPTION
**Pull request by timj**

### CODE REVIEW

File: python/lsst/source/injection/inject_base.py [hunk 1]
```diff
@@ -334,11 +334,11 @@ def _compose_injection_catalog(self, injection_catalogs):
...
-            dict(
-                injection_id=injection_id,
-                injection_flag=np.zeros(len(injection_data), dtype=int),
-                injection_draw_size=np.zeros(len(injection_data), dtype=int),
-            )
+            {
+                "injection_id": injection_id,
+                "injection_flag": np.zeros(len(injection_data), dtype=int),
+                "injection_draw_size": np.zeros(len(injection_data), dtype=int),
+            }
```
Comment (leeskelvin) [inline python/lsst/source/injection/inject_base.py:341]: This is obviously fine, but why the change on this ticket?

File: python/lsst/source/injection/inject_engine.py [hunk 2]
```diff
@@ -25,7 +25,8 @@
 
 import os
 from collections import Counter
-from typing import Any, Generator
+from collections.abc import Generator
+from typing import Any
```
Comment (leeskelvin) [inline python/lsst/source/injection/inject_engine.py:29]: As below with `collections.abc.Sequence`, is there a good reason to favour one over the other here?

**Review by leeskelvin — APPROVED** [review]

File: python/lsst/source/injection/inject_engine.py (hunk 2 above)
Comment (timj) [reply python/lsst/source/injection/inject_engine.py:29]: Anything that is in `collections.abc` is now always preferred and the `typing` variant is deprecated.

Comment (leeskelvin) [reply python/lsst/source/injection/inject_engine.py:29]: (see @timj's reply above, thanks :+1:)
